src/app/Http/Controllers/Operations/ListOperation.php

```diff
@@ -47,7 +47,7 @@ public function search()
         }
         // limit the number of results according to the datatables pagination
         if ($this->request->input('length')) {
-            $this->crud->take($this->request->input('length'));
```

Comment:
Totally agree with this.

Reply:
All I did to that line was cast the length as an integer, as otherwise it is cast as a string by default and this is a problem with Mongo but apparently not a problem with Eloquent

```
$this->crud->take((int)$this->request->input('length'));
```

What is your concern with that?